src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -301,7 +301,8 @@ ucs_status_t uct_ib_mlx5_txwq_init(uct_priv_worker_t *worker,
 
     status = uct_ib_mlx5_get_qp_info(verbs_qp, &qp_info);
     if (status != UCS_OK) {
-        ucs_error("Failed to get mlx5 QP information");
+        ucs_error("Failed to get mlx5 QP information. Please make sure that the"
```

Comment:
1. pls have a common function to print the error message to avoid text duplication
  uct_ib_mlx5_obj_error(const char *obj_name) (obj_name="cq"/"qp"/"srq")
1. pls move the error message inside ib_mlx5.c and print it in the following functions: get_qp_info, get_cq_info, get_srq_info, in the code which is under '#if HAVE_DECL_IBV_MLX5_EXP_GET_xx_INFO' (not the #else). This is in order to fix the code to adhere to the error propagation convention: the error should be printed when the ucs_status_t code is generated.
1. the error should be "Failed to get mlx5 %s information. Please make sure the installed libmlx5 version matches the one UCX was compiled with."
1. maybe we can try to run `rpm -qf /usr/include/infiniband/mlx5_hw.h` during compilation time, and if it works, embed the result in the code (as a preprocessor macro defined from the command line), so we can display to the user the libmlx5 version UCX was compiled with?